pkg/build/linux.go

```diff
@@ -65,7 +65,21 @@ func (linux) buildKernel(params *Params) error {
 	case "ppc64le":
 		target = "zImage"
 	}
-	if err := runMake(params.KernelDir, target, "CC="+params.Compiler); err != nil {
+
+	ccParam := params.Compiler
+	if params.Ccache != "" {
+		ccParam = params.Ccache + " " + ccParam
+		// Ensure CONFIG_GCC_PLUGIN_RANDSTRUCT doesn't prevent ccache usage.
+		// See /Documentation/kbuild/reproducible-builds.rst.
+		err := osutil.WriteFile(filepath.Join(params.KernelDir, "scripts", "gcc-plugins",
```

Comment:
We test kernel at least back to 4.1 and I think some users even use it for 3.18. And bisection can also go back in history.
I see on 4.1 there is no scripts/gcc-plugins, so this will fail build. I think it's safer to check if scripts/gcc-plugins exsts with osutil.IsExist and write the file only then. It will also avoid the need for the change in pkg/vcs/linux_test.go.

Reply:
Fixed now as suggested